src/block_disagg/block_disagg_ckpt.c

```diff
@@ -0,0 +1,227 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+/*
+ * __bmd_checkpoint_pack_raw --
+ *     This function needs to do two things: Create a recovery point in the object store underlying
+ *     this table and create an address cookie that is saved to the metadata (and used to find the
+ *     checkpoint again).
+ */
+static int
+__bmd_checkpoint_pack_raw(WT_BLOCK_DISAGG *block_disagg, WT_SESSION_IMPL *session,
+  WT_ITEM *root_image, WT_PAGE_BLOCK_META *block_meta, WT_CKPT *ckpt)
+{
+    uint32_t checksum, size;
+    uint8_t *endp;
+
+    WT_ASSERT(session, block_meta != NULL);
+    WT_ASSERT(session, block_meta->page_id != WT_BLOCK_INVALID_PAGE_ID);
+
+    /*
+     * !!!
+     * Our caller wants the final checkpoint size. Setting the size here violates layering,
+     * but the alternative is a call for the btree layer to crack the checkpoint cookie into
+     * its components, and that's a fair amount of work.
+     */
+    ckpt->size = block_meta->page_id; /* XXX What should be the checkpoint size? Do we need it? */
+
+    /*
+     * Write the root page out, and get back the address information for that page which will be
+     * written into the block manager checkpoint cookie.
+     *
+     * TODO: we need to check with the page service team if we need to write an empty root page.
```

Comment:
@quchenhao It looks like we can resolve this in the affirmative? We need the `disagg_lsn` from the `block_meta` structure for `__wt_block_disagg_ckpt_pack` to work. I plan to remove this `TODO` for the disagg block manager PR.